pandas/core/arrays/period.py

```diff
@@ -51,6 +51,7 @@ def _period_array_cmp(cls, op):
     opname = '__{name}__'.format(name=op.__name__)
     nat_result = True if opname == '__ne__' else False
 
+    # @ops.unwrap_and_defer
```

Comment:
@TomAugspurger any idea why not returning NotImplemented for DataFrame?

Reply:
Not sure... I vaguely remember a broadcasting error, but that may have been user error.